### DISCUSSION TITLE
Linux Mint should replace certain built-in driver packages with better alternatives & add additional software

### DISCUSSION THREAD

**Discussion by FrozenDeathKnight** [linuxmint/discussions#559] [Ideas]
This is kind of a review after having used Linux Mint 21.3 for well over a month and recently installed V22.

For the most part I have been very happy with the way Mint works out of the box. That being said, I have had my fair share of hurdles that I have had to fix to make it fully operational. There are a few software packages on Mint that honestly could be replaced with better alternatives when getting the hardware to work out of the box. Here's the list:

1. OpenTabletDriver replacing Wacom Graphics Tablet. After having tried both I can safely say that OpenTabletDriver is the superior alternative over the built-in tablet driver, especially in regards to dealing with multiple monitors, shortcut options, and tech support (there is a dedicated Discord that has excellent tech support for setting up your tablet driver unlike the Wacom one). The current one works, but there are tons of missing features I have come to expect from a tablet driver that just aren't present with Wacom's open source Linux drivers. It just doesn't make sense to ship something that is just inferior to something that is built for Linux, Windows, and Mac. A compromise would be to choose which one to install during installation of the OS, but overall I think OpenTabletDriver is just the much better option.
2. Pulseaudio Volume Control replacing Sound. It's basically a must have if you deal with audio production and external audio interfaces like I do. The current built-in Sound settings can't even recognise my Roland Super UA-S10 when I have it plugged in using Pipewire (another issue that thankfully has been fixed in V22). Pulseaudio Volume Control has none of these issues and I can even adjust individual channels and use different audio sources for different programs at the same time. I never use the built-in audio settings for this very reason. The only negative thing about is that it doesn't respect my desktop dark mode settings, but if it works it works.

Here are a few other software that I also consider a must have for a better baseline experience:

1. Flatseal. The GUI is great and it makes dealing with Flatpaks a breeze.
2. Wine/Bottles + Winetricks. It just makes sense to have this as baseline considering how many still rely on Windows based applications.
3. QJackCtl + qpwgraph. Any type of advanced audio basically requires both of these to function. Jack is needed for low latency recording for specific DAWs like Ardour and the graph for observing the audio channels are essential for seeing potential issues with audio input and output when troubleshooting. Having both being more integrated would do a lot to make Linux Mint really solid for any audio related tasks. Pipewire is a solid first step, but we need more audio tools to be readily available and supported out of the box.
4. LibreOffice (option during install). Just makes sense to have the option to install these immediately when Mint is meant to have as close of a Windows like experience as possible.
5. Psensor should be built into the System Monitor itself. Having statistics on your hardware like heat is honestly very important when checking if something is wrong with your PC.
6. Meshlab or F3D. Having a way to open 3D files is honestly pretty important, so having a baseline way of doing so would be great.
7. Mintupgrade. Didn't know this existed until I got told about installing it through the Terminal. Seems a bit odd that this isn't built-in as a part of the GUI of Mint when the philosophy is to have a graphic user interface for the essentials.
8. Rclone type functionality with Online Accounts. Being able to have my Google Drive available physically on my disk beyond just having access to the folders through Nemo would make Online Accounts just way more useful for production work. Being unable to sync physically to my drives and feed that back to my cloud makes it a lot harder to work with other people or when doing cloud based projects.
9. RabbitVCS integration with Nemo to be able to work with SVN, Github, and Hg with my physical drives. It's the only file explorer that requires manual installation of the .py-file extension from Github to be able to work with SVN servers, which are also commonly used for team based productions.

There may be other software as well, but these I think are the most essential for having a more well rounded, out of the box experience. Some are just basic baseline installations while others are more about integrating the functionality into what already exists.

**Comment by Jopp-gh** [comment]
A reasonable list!

**opentablet** driver: agreed, a upgraded tablet driver is absolutely needed. I'm missing the ability to have profiles for a variety of creative apps. (Krita, blender, etc). The default wacom driver in Mint works great but comes without profiles. Profiles are just some pref files, they shouldn't be a pain to integrate. It seems, most people don't use tablets, so there's no outcry of something missing like trivial profiles.

I tried hard to get the opentablet driver to work but with little success. (Xppen star 3) Also, I  sent a bug report and got a pretty good support to setup the old (I guess, deb) package. Unlucky this worked only partially. After some trouble I just uninstalled this driver. The flatpak then, doesn't work at all. 

I'm playing with the idea to write my own Wacom profile switcher.

1. Flatseal and I'd add Warehouse too
2. Wine bottles might be a good suggestion, I heard  bottles to be safer. However I'm not sure if it is superior to Lutris in terms of security. Currently I'm using Lutris and everything works fine.
3. Libre Office - a must if you want to create presentations or tables. Indispensable, imo.
4. Psensor is great albeit not always exact to measure degrees C. But indispensable to keep fan temps under control, so in addition, I'd suggest **Fan-control** too
5. Meshlab - need to check it out 
6. Mintupgrade ? cool, as  tool. Maybe not essential but comfortable. BTW, OS updates aren't always this essential, if not for technical details (support for newer hw components)
7. I don't use audio tools that much, so I cant add any thought in regards

**Reply by FrozenDeathKnight** [reply]
@Jopp-gh I actually wrote a tutorial on how to install OpenTabletDriver for Mint 22 on the Linux Mint forums:

https://forums.linuxmint.com/viewtopic.php?p=2502266#p2502266

The Flatpak is not officially supported by the people making the driver and I was recommended not to use that one.

Would be nice to have profiles per software like you can on Windows with the Wacom drivers. However, unless Wacom decide to fully support Linux by actually putting in the work to upgrade the open source driver to be equal in features to the other operating systems, I would recommend using OpenTabletDriver instead. For multiple reasons.
1. It has the vast majority of features you would expect from the proprietary Wacom drivers.
2. It is open source and is not reliant on Wacom to fix issues with it.
3. It is already OS agnostic as in it being being used on Windows, Mac, and Linux.
4. It has a dedicated community that relies on the driver. like the people who play OSU.
5. The driver is stable after having run it on both 21.3 and 22. As long as no other drivers are causing conflicts with OTD it runs well. The only issue I found was that having one screen being off before logging in to Mint can cause a weird issues if you activate it after, but that might be more related to Linux itself and not running Wayland with Mint.

Also, having a Linux OS like Mint officially support the driver as a part of the OS would help make the project stronger in the future. The OTD devs have done a great job developing, maintaining, documenting, and giving out tech support. I think such work should be recognised by having it as an officially supported driver for Linux Mint as an alternative to running the Wacom open source driver.

**Reply by Jopp-gh** [reply]
>  actually wrote a tutorial on how to install OpenTabletDriver for Mint 22

Nice, today I tried again **open tablet driver** and installed their latest version with a '*.deb' package. Everything worked out of the box this time.

I still added the startup cmd, though.

Well, OTD lets you save presets, but I didn't understand how to switch between presets via shortcut. Usually, a tablet driver should to this switch automatically, based on the active app in use.

Because profiles is really the only feature I'm missing with the default installation. Probably you use some OTD plugin to switch profiles? 

Just one profile might work well for occasional use, but not if you've a bunch of creative apps with different keyboard shortcuts. 

Agreed, open source would be definitely a better choice than Wacom drivers.

Time ago, I added a request to modernize input devices
https://github.com/orgs/linuxmint/discussions/472
